src/components/application_manager/src/hmi_language_handler.cc

```diff
@@ -274,13 +278,21 @@ void HMILanguageHandler::VerifyWithPersistedLanguages() {
 
 void HMILanguageHandler::HandleWrongLanguageApp(const Apps::value_type& app) {
   SDL_AUTO_TRACE();
-  Apps::iterator it = apps_.find(app.first);
-  if (apps_.end() == it) {
-    SDL_DEBUG("Application id "
-              << app.first << " is not found within apps with wrong language.");
-    return;
+  {
+    sync_primitives::AutoLock lock(apps_lock_);
+    Apps::iterator it = apps_.find(app.first);
+    if (apps_.end() == it) {
+      SDL_DEBUG("Application id "
+                << app.first
+                << " is not found within apps with wrong language.");
+      return;
+    }
+    apps_.erase(it);
+    if (0 == apps_.size()) {
```

Comment:
@okozlovlux Can you please consider using .empty() instead of comparing to 0. It's much more effective.